core/src/main/java/org/apache/accumulo/fate/Fate.java

```diff
@@ -226,11 +224,26 @@ public Fate(T environment, TStore<T> store, Function<Repo<T>,String> toLogStrFun
    * Launches the specified number of worker threads.
    */
   public void startTransactionRunners(AccumuloConfiguration conf) {
-    int numThreads = conf.getCount(Property.MANAGER_FATE_THREADPOOL_SIZE);
-    executor = ThreadPools.createExecutorService(conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
-    for (int i = 0; i < numThreads; i++) {
-      executor.execute(new TransactionRunner());
-    }
+    final ThreadPoolExecutor pool = (ThreadPoolExecutor) ThreadPools.createExecutorService(conf,
+        Property.MANAGER_FATE_THREADPOOL_SIZE);
+    fatePoolWatcher = ThreadPools.createGeneralScheduledExecutorService(conf);
+    fatePoolWatcher.schedule(() -> {
+      ThreadPools.resizePool(pool, conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
+      // Assume a thread could execute up to 100 operations a second.
+      // We are sleeping for three seconds. So to calculate max to queue do 3* 100 * numThreads.
+      int maxToQueue = 300 * conf.getCount(Property.MANAGER_FATE_THREADPOOL_SIZE);
+      int remaining = maxToQueue - pool.getQueue().size();
+      for (int i = 0; i < remaining; i++) {
+        try {
+          pool.execute(new TransactionRunner());
+        } catch (RejectedExecutionException e) {
+          // RejectedExecutionException could be shutting down
+          log.warn("Error adding transaction runner to FaTE executor pool.");
```

Comment:
I think if its shutdown we can avoid spamming the logs.  If its not shutdown, then its really bad and we should include the exception.  For the case where its not shutdown, should we log an error or rethrow the exception?

```suggestion
      if(pool.isShutdown()){
          // The exception is expected in this case, no need to spam the logs.
          log.trace("Error adding transaction runner to FaTE executor pool.",e);
       } else {
        // This is bad, FaTE may no longer work!
         log.error("Error adding transaction runner to FaTE executor pool.",e);
       }
```